pkg/ansible/generate.go

```diff
@@ -332,21 +338,77 @@ func GenerateClusterWideVars(name string, hardwareSpec *coapi.ClusterHardwareSpe
 	return buf.String(), nil
 }
 
+func lookupAMIForMachineSet(machineSet *coapi.MachineSet, clusterVersion *coapi.ClusterVersion) (string, error) {
+	for _, regionAMI := range clusterVersion.Spec.VMImages.AWSImages.RegionAMIs {
+		if regionAMI.Region == machineSet.Spec.ClusterHardware.AWS.Region {
+			if machineSet.Spec.MachineSetConfig.NodeType == coapi.NodeTypeMaster && regionAMI.MasterAMI != nil {
+				return *regionAMI.MasterAMI, nil
+			}
+			return regionAMI.AMI, nil
+		}
+	}
+	return "", fmt.Errorf("no AMI defined for cluster version %s/%s in region %v", clusterVersion.Namespace, clusterVersion.Name, machineSet.Spec.ClusterHardware.AWS.Region)
```

Comment:
This is a good error here. But in addition to this, we need to have cluster controller block creation of machine sets when the cluster is for a region that is not included in the cluster version. The cluster should also have a condition added in that case so that the user can see that there is a problem that they need to address.